api/v1/submissions/PKPSubmissionHandler.inc.php

```diff
@@ -657,6 +658,13 @@ public function editPublication($slimRequest, $response, $args) {
 			return $response->withStatus(403)->withJsonError('api.publication.403.cantEditPublished');
 		}
 
+		// Prevent users from editing publications if they do not have permission. Except for admins.
+		$userRoles = $this->getAuthorizedContextObject(ASSOC_TYPE_USER_ROLES);
+		$canEditPublication = !empty(array_intersect(array(ROLE_ID_SITE_ADMIN), $userRoles));
+		if (!$canEditPublication && !Services::get('submission')->canUserEditMetadata($submission->getId(), $currentUser->getId())) {
```

Comment:
I think we can get rid of `$canEditPublication` with:

```
if (!in_array(ROLE_ID_SITE_ADMIN, $userRoles) && !Services::get(...)) {
```

Reply:
check